### PR TITLE
Switch controller-manager to logr (3)

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area logging dev-productivity
/kind enhancement

**What this PR does / why we need it**:

Continuation of https://github.com/gardener/gardener/pull/5268
Migrate more controller-manager controllers:
- event controller
- exposureclass controller
- plant controller
- managedseedset controller

Also extend the logging guideline with some new learnings (first commit, to make the other commits more understandable).

**Which issue(s) this PR fixes**:
Part of #4251

**Special notes for your reviewer**:

/squash

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/controllermanager/controller/bastion/bastion_control.go [hunk 1]
```diff
@@ -156,7 +156,7 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 	// bastions for it, so this check here is just a safety measure.
 	if !equality.Semantic.DeepEqual(shoot.Spec.SeedName, bastion.Spec.SeedName) {
 		log.Info("Deleting bastion because the referenced Shoot has been migrated to another Seed",
-			"oldSeed", bastion.Spec.SeedName, "newSeed", shoot.Spec.SeedName)
+			"oldSeedNew", bastion.Spec.SeedName, "newSeedName", shoot.Spec.SeedName)
```
Comment (rfranzke) [inline pkg/controllermanager/controller/bastion/bastion_control.go]: This should be `oldSeedName`, or?

File: pkg/controllermanager/controller/plant/plant_control.go [hunk 2]
```diff
@@ -19,46 +19,45 @@ import (
...
+		c.log.Error(err, "Could not list plants for secret", "secret", client.ObjectKeyFromObject(secret))
 		return
 	}
 
 	for _, plant := range plantList.Items {
 		if isPlantSecret(plant, kutil.Key(secret.Namespace, secret.Name)) {
 			key, err := cache.MetaNamespaceKeyFunc(&plant)
 			if err != nil {
-				logger.Logger.Errorf("Couldn't get key for plant %+v: %v", plant, err)
+				c.log.Error(err, "Couldn't get key for object", "object", obj)
```
Comment (rfranzke) [inline pkg/controllermanager/controller/plant/plant_control.go]: Shouldn't this be "plant" and not "object"?

**Comment by timebertt** [comment]
Thanks, @rfranzke. Nice catches :)
PTAL

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**